storage/shard.go

```diff
@@ -271,18 +274,22 @@ func (s *dbShard) Bootstrap(bs m3db.Bootstrap, writeStart time.Time, cutover tim
 	s.bs = bootstrapping
 	s.Unlock()
 
+	multiErr := xerrors.NewMultiError()
 	sr, err := bs.Run(writeStart, s.shard)
 	if err != nil {
-		return err
+		renamedErr := fmt.Errorf("error occurred bootstrapping shard %d from external sources: %v", s.shard, err)
+		err = xerrors.NewRenamedError(err, renamedErr)
 	}
-
-	bootstrappedSeries := sr.GetAllSeries()
-	for id, dbBlocks := range bootstrappedSeries {
-		series, completionFn := s.writableSeries(id)
-		err := series.Bootstrap(dbBlocks, cutover)
-		completionFn()
-		if err != nil {
-			return err
+	multiErr = multiErr.Add(err)
```

Comment:
Perhaps best to add to `multiErr` inside the `err != nil` clause?  Semantically seems more right.